include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -64,20 +67,23 @@ struct __scan_status_flag
         for (int tile = static_cast<int>(tile_id) + offset; tile >= 0; tile -= 32)
         {
             _AtomicRefT tile_atomic(*(flags_begin + tile + padding - local_id));
-            std::uint32_t tile_val = 0;
-            do {
-                tile_val = tile_atomic.load();
+            std::uint32_t flag = NOT_READY;
```

Comment:
`flag` doesn't need to be initialized here.

Reply:
Removed initialization.